pennylane/devices/preprocess.py

```diff
@@ -155,8 +155,9 @@ def mid_circuit_measurements(
 
     If the tape of device uses finite-shot, use the native implementation (i.e. no transform), and use defer measurements transforms otherwise.
     """
+
     if tape.shots and tape.batch_size is None:
-        return (tape,), null_postprocessing
+        return qml.dynamic_one_shot(tape)
```

Comment:
i might be missing something obvious, but where was the entrypoint to all this native MCM code before this change?

Reply:
It used to be in `simulate.py` [here](https://github.com/PennyLaneAI/pennylane/pull/5266/files/7dea3bba6f38351b8c1f51a71d14cd55fb993236#diff-a5e8e7161dde1247b6305745302f216f95e68ea8e42ceb7dea59bae63cf2b263L267). Now we bypass the main native function (which is essentially replaced by `dynamic_one_shot` postprocessing and directly call `simulate_one_shot_native_mcm`.

Reply:
sorry, meant to ask where we were calling it from. Now I see it was line 257, and we now just call the other one directly as you said. looks good!